### PR TITLE
fix(buildkite): correctly parse plugins that are quoted

### PR DESCRIPTION
**Pull request by jamietanna**
## Changes

It was noticed that when there are quotes around Buildkite plugin usages,
Renovate would include the quote at the start of the `depName` and the
tail of the `currentValue`.

This makes the regex a little less greedy by excluding either single or
double quotes.

## Context

It was noticed that when there are quotes around Buildkite plugin usages,
Renovate would include the quote at the start of the `depName` and the
tail of the `currentValue`.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/buildkite/__fixtures__/pipeline10.yml
Comment (secustor) [inline lib/modules/manager/buildkite/__fixtures__/pipeline10.yml:1]: Please inline using a `codeBlock`

**Review by secustor — COMMENTED** [review]
else LGTM

File: lib/modules/manager/buildkite/__fixtures__/pipeline10.yml
Comment (jamietanna) [reply lib/modules/manager/buildkite/__fixtures__/pipeline10.yml:1]: I'd left it to follow the existing patterns in the file - happy to inline, if we're happy it being inconsistent?

Comment (rarkins) [reply lib/modules/manager/buildkite/__fixtures__/pipeline10.yml:1]: We believe it to be more efficient (speed) plus more readable, so we want future tests to inline using codeBlock whenever feasible

Comment (viceice) [reply lib/modules/manager/buildkite/__fixtures__/pipeline10.yml:1]: also dependabot doesn't find those fixtures and can't report false alarms.

Comment (jamietanna) [reply lib/modules/manager/buildkite/__fixtures__/pipeline10.yml:1]: All good - want me to do a prefactor PR to replace at least this file's fixtures with codeBlock?

(I'm also happy to raise one for everything in the codebase if it feels straightforward)

Comment (rarkins) [reply lib/modules/manager/buildkite/__fixtures__/pipeline10.yml:1]: Either would be great!

(I wonder if any of the AI's can do this with a codebase)

Comment (jamietanna) [reply lib/modules/manager/buildkite/__fixtures__/pipeline10.yml:1]: https://github.com/renovatebot/renovate/pull/31134 (for just the Buildkite side of things so far)

**Comment by jamietanna** [comment]
Oops, sorry I rebased + force pushed :sweat_smile:

**Review by rarkins — APPROVED** [review]

**Pull request merged**